RiotSwiftUI/Modules/Authentication/Common/Service/MatrixSDK/AuthenticationService.swift

```diff
@@ -73,7 +77,39 @@ class AuthenticationService: NSObject {
     }
     
     // MARK: - Public
-    
+
+    /// Parse and handle a server provisioning link.
+    /// - Parameter universalLink: A link such as https://mobile.element.io/?hs_url=matrix.example.com&is_url=identity.example.com
+    /// - Returns: `true` if a provisioning link was detected and handled.
+    @discardableResult
+    func handleServerProvisioningLink(_ universalLink: UniversalLink) -> Bool {
+        MXLog.debug("[AuthenticationService] handleServerProvisioningLink: \(universalLink)")
+
+        let hsUrl = universalLink.homeserverUrl
+        let isUrl = universalLink.identityServerUrl
+
+        if hsUrl == nil && isUrl == nil {
+            MXLog.debug("[AuthenticationService] handleServerProvisioningLink: no hsUrl or isUrl")
+            return false
+        }
+
+        let isRegister = universalLink.pathParams.first == "register"
+        let flow: AuthenticationFlow = isRegister ? .register : .login
+
+        if needsAuthentication {
+            //  not logged in
+            //  update the state with given HS and IS addresses
+            state = AuthenticationState(flow: flow,
```

Comment:
This should probably call `reset()` first as the wizards (if created) will not match up to this new server.

Reply:
👍